genomepy/functions.py

```diff
@@ -499,6 +499,13 @@ def gap_sizes(self):
         """
         if not self._gap_sizes:
             gap_file = self.props["gaps"]["gaps"]
+            
+            # generate gap file if not found
+            if not os.path.exists(gap_file):
+                from genomepy.utils import generate_gap_bed
+                genome_fasta = os.path.join(os.path.dirname(gap_file), self.name)
```

Comment:
You can use `self.filename`, this will contain the full path to the genome FASTA file.